koin-projects/koin-core/src/main/kotlin/org/koin/core/scope/ScopeDefinition.kt

```diff
@@ -46,7 +46,8 @@ data class ScopeDefinition(val scopeName: String, val module: Module) {
     }
 
     internal fun release(instance: ScopeInstance) {
-        definitions.filter { it is ScopedInstance<*> }.forEach { it.instance.release(InstanceContext(scope = instance)) }
+        definitions.mapNotNull { it.instance as? ScopedInstance<*> }
```

Comment:
why not filter operator here?

Reply:
Saved an extra instance call. But I agree, filter would be more readable here.